src/PostgREST/Error.hs

```diff
@@ -19,10 +21,17 @@ import           Network.Wai               (Response, responseLBS)
 errResponse :: HT.Status -> Text -> Response
 errResponse status message = responseLBS status [(hContentType, "application/json")] (cs $ T.concat ["{\"message\":\"",message,"\"}"])
 
-pgErrResponse :: H.Error -> Response
+pgErrResponse :: P.UsageError -> Response
 pgErrResponse e = responseLBS (httpStatus e)
   [(hContentType, "application/json")] (JSON.encode e)
 
+instance JSON.ToJSON P.UsageError where
+  toJSON (P.ConnectionError e) = JSON.object [
+    "code" .= ("" :: T.Text),
+    "message" .= ("Connection error" :: T.Text),
+    "details" .= (cs (fromMaybe "" e) :: T.Text)]
+  toJSON e = JSON.toJSON e -- H.Error
```

Comment:
The locking comes from the line above

change
```toJSON e = JSON.toJSON e -- H.Error```
to
```toJSON (P.SessionError e) = JSON.toJSON e -- H.Error```

after this the tests will not lock but most of them will fail with "current transaction is aborted, commands ignored until end of transaction block".
that is because on the db connection, a previous test executed a failing query (missing table) and the transaction was not automatically rolled back. Looking if i can pinpoint where exactly